---
title: |
  ![](flomics_icon.png){width=3in}
  <p style="color:#00AEEF">________________________________________________________</p>

  <p style="color:#00AEEF">RNAseq report from `r sample`</p>
date: <h4 style="font-style:normal"><p style="color:#00AEEF">`r Sys.Date()`</p></h4><br />
output:
  html_document:
    toc: true
    toc_float: true
    theme: cerulean
keep_md: true
sample: "`r sample`"
single_end: "`r single_end`"
genome: "`r genome`"
gtf: "`r gtf`"
gencode: "`r gencode`"
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
library(knitr)
library(OpenImageR)
library(kableExtra)
```

```{css, echo = FALSE}
h1 {
  color: #00AEEF;
  font-weight: bold;
  font-size: 24px;
}
h2 {
  color: #00AEEF;
  font-weight: bold;
  font-size: 18px;
}
```


#  General analyses

\


## Parameters

```{r echo = FALSE}
if (genome != "") {
  parameters <- data.frame(parameter = c("Genome version", "Annotation version (Ensembl)", "Gencode version"), Value = c(genome, gtf, gencode))
  kable(parameters, align = 'l') %>%
    kable_styling(full_width = F) %>%
    column_spec(1, width = "50em") %>%
    column_spec(2, width = "30em")
}
```

\


##  Number of reads

```{r echo = FALSE}
```{r echo = FALSE}
tabseq <- paste("fastqc/", sample, "_total_sequences.txt", sep = "")
seqpath <- file.path(getwd(), tabseq)
if (file.exists(seqpath)){
  seqs <- read.table(seqpath, stringsAsFactors = FALSE, sep = "\t")
  seqs[,1] <- gsub(" .+", "", seqs[,1])
  if (single_end == "paired") {
    total <- data.frame(sample_name = c("R1", "R2"), Sequences = c(as.numeric(seqs[2,1])/4, as.numeric(seqs[2,2])/4))
    colnames(total)[1] <- sample
  } else {
    total <- data.frame(sample_name = c("single_end"), Sequences = c(as.numeric(seqs[1,1]))/4)
    colnames(total)[1] <- sample
  }
  kable(total) %>%
    kable_styling(full_width = F) %>%
    column_spec(1, width = "50em") %>%
    column_spec(2, width = "30em")
} else {
  message ("Step fastqc was skipped")
}
```

\


##  Number of trimmed reads

```{r echo = FALSE}
tabtrim <- paste("trim_galore/", sample, "_after_trimming_sequences.txt", sep = "")
path <- file.path(getwd(), tabtrim)
if (file.exists(path)){
  trimmed <- read.table(path, sep = "\t", stringsAsFactors = FALSE)
  trimmed$V2 <- gsub(".f.+q.gz","",trimmed$V2,ignore.case=T)
  trimmed$V1 <- gsub("Filename","Sample",trimmed$V1,ignore.case=T)
  numseqfile <- paste("trim_galore/", sample, "_number_sequences.txt", sep = "")
  numseqpath <- file.path(getwd(), numseqfile)
  numseq <- read.table(numseqpath, stringsAsFactors = FALSE, sep = "\t")
  if (single_end == "paired") {
    trim <- data.frame(sample_name = c("R1", "R2"), Discarded_reads = c(as.numeric(numseq[1,1])/4-as.numeric(trimmed[2,2]), as.numeric(numseq[2,1])/4-as.numeric(trimmed[4,2])))
    colnames(trim)[1] <- sample
  } else {
    trim <- data.frame(sample_name = c("single_end"), Discarded_reads = c(as.numeric(numseq[1,1])/4-as.numeric(trimmed[2,2])))
    colnames(trim)[1] <- sample
  }
  kable(trim) %>%
    kable_styling(full_width = F) %>%
    column_spec(1, width = "50em") %>%
    column_spec(2, width = "30em")
} else {
  message ("Step for adapter trimming has been skipped")
}

```

\


## Number of mapped reads

```{r echo = FALSE}
if (single_end == "paired") {
  mapfile <- paste("star/", sample, "_1Log.final.out", sep = "")
} else {
  mapfile <- paste("star/", sample, "Log.final.out", sep = "")
}
mappath <- file.path(getwd(), mapfile)
if (file.exists(mappath)){
  mapped <- read.table(mappath, stringsAsFactors = FALSE, sep = "\\")
  input <- as.numeric(gsub("[A-Za-z]+|\t|\\|| ", "", mapped[5,]))
  mapped <- as.numeric(gsub("[A-Za-z]+|\t|\\|| ", "", mapped[8,])) + as.numeric(gsub("[A-Za-z]+|\t|\\|| ", "", mapped[23,])) + as.numeric(gsub("[A-Za-z]+|\t|\\|| ", "", mapped[25,]))
  mapped_percentage <- round(mapped/input * 100, 2)
  if (single_end == "paired") {
    total_reads <- input*2
    mapped_reads <- mapped*2
  } else {
    total_reads <- input
    mapped_reads <- mapped
  }
  maptab <- data.frame(names = c("Total reads for mapping", "%mapped reads", "mapped reads"), Values = c(as.integer(total_reads), mapped_percentage, as.integer(mapped_reads)))
  kable(maptab, row.names = NA, col.names = c("", "value"))  %>%
    kable_styling(full_width = F) %>%
    column_spec(1, width = "50em") %>%
    column_spec(2, width = "30em")
} else {
  message("Mapping step has been skipped")
}

```


\


## Transcript coverage profile

```{r echo = FALSE}
file <- paste("qualimap/", sample, ".coverage.txt", sep = "")
coverage <- read.table(file, header = FALSE, stringsAsFactors = FALSE)
coverage.df <- data.frame(bias = coverage$V1, value = coverage$V4)
kable(coverage.df, row.names = NA, col.names = c("Bias", "Value")) %>%
  kable_styling(full_width = F) %>%
  column_spec(1, width = "50em") %>%
  column_spec(2, width = "30em")
```

```{r echo = FALSE}
im_file <- paste("qualimap/", sample, ".genome_coverage_across_reference.png", sep = "")
path <- file.path(getwd(), im_file)
im <- readImage(path)
imageShow(im)
```


\


## Number of reads per gene biotype

```{r echo = FALSE}
file <- paste("features/", sample, "Aligned.sortedByCoord.out_biotype_counts_mqc.txt", sep = "")
biotypes <- read.table(file, header = FALSE, stringsAsFactors = FALSE)
#coverage.df <- data.frame(bias = coverage$V1, value = coverage$V4)
kable(biotypes, row.names = NA, col.names = c("Biotype", "Number of Reads")) %>%
  kable_styling(full_width = F) %>%
  column_spec(1, width = "50em") %>%
  column_spec(2, width = "30em")
```


\


# Contact

<font size="3">If you have any question, please contact [support@flomics.com](mailto:support@flomics.com) </font>
